x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/use_ti_data_sources.ts

```diff
@@ -0,0 +1,166 @@
+  data.search.search<CtiDataSourceRequestOptions, CtiDataSourceStrategyResponse>(
+    {
+      defaultIndex,
+      factoryQueryType: CtiQueries.dataSource,
+      timerange,
+    },
+    {
+      strategy: 'securitySolutionSearchStrategy',
+      abortSignal: signal,
+    }
+  );
+
+export const getTiDataSourcesComplete = (
+  props: GetThreatIntelSourcProps
+): Observable<CtiDataSourceStrategyResponse> => {
+  return getTiDataSources(props).pipe(
+    filter((response) => {
+      return isErrorResponse(response) || isCompleteResponse(response);
+    })
+  );
+};
+
+const getTiDataSourcesWithOptionalSignal = withOptionalSignal(getTiDataSourcesComplete);
+
+export const useTiDataSourcesComplete = () => useObservable(getTiDataSourcesWithOptionalSignal);
+
+export interface TiDataSources {
+  dataset: string;
+  name: string;
+  count: number;
+  dashboardId?: string;
+}
+interface TiDataSourcesProps extends Partial<GlobalTimeArgs> {
+  allTiDataSources?: TiDataSources[];
+}
+
+export const useTiDataSources = ({
+  to,
+  from,
+  allTiDataSources,
+  setQuery,
+  deleteQuery,
+}: TiDataSourcesProps) => {
+  const [tiDataSources, setTiDataSources] = useState<TiDataSources[]>([]);
+  const { data, uiSettings } = useKibana().services;
+  const defaultThreatIndices = uiSettings.get<string[]>(DEFAULT_THREAT_INDEX_KEY);
+  const { result, start, loading } = useTiDataSourcesComplete();
+
+  useEffect(() => {
+    start({
+      data,
+      timerange: to && from ? { to, from, interval: '' } : undefined,
+      defaultIndex: defaultThreatIndices,
+    });
+  }, [to, from, start, data, defaultThreatIndices]);
+
+  useEffect(() => {
+    if (!loading && result && result?.rawResponse && result?.inspect && setQuery) {
+      setQuery({
+        id: ID,
+        inspect: {
+          dsl: result?.inspect?.dsl ?? [],
+          response: [JSON.stringify(result.rawResponse, null, 2)],
+        },
+        loading,
+        refetch: () => {},
+      });
+    }
+  }, [setQuery, loading, result]);
+
+  useEffect(() => {
+    return () => {
+      if (deleteQuery) {
+        deleteQuery({ id: ID });
+      }
+    };
+  }, [deleteQuery]);
+
+  useEffect(() => {
+    if (!loading && result) {
+      const datasets = result?.rawResponse?.aggregations?.dataset?.buckets ?? [];
+      const getChildAggregationValue = (aggregation?: Bucket) => aggregation?.buckets?.[0]?.key;
+
+      const integrationMap = datasets.reduce((acc: Record<string, TiDataSources>, dataset) => {
+        const datasetName = getChildAggregationValue(dataset?.name);
+        if (datasetName) {
+          return {
+            ...acc,
+            [dataset.key]: {
+              dataset: dataset?.key,
+              name: datasetName,
+              dashboardId: getChildAggregationValue(dataset?.dashboard),
+              count: dataset?.doc_count,
+            },
+          };
+        } else {
+          const otherTiDatasetKey = '_others_ti_';
```

Comment:
this could probably live in the CTI constants file

Reply:
fixed